Jenkinsfile

```diff
@@ -0,0 +1,49 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2019 Joyent, Inc.
+ */
+
+@Library('jenkins-joylib@v1.0.2') _
+
+pipeline {
+
+    agent {
+        label joyCommonLabels(image_ver: '1.6.3', pkgsrc_arch: 'i386')
+    }
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+    }
+
+    stages {
+        stage('check') {
+            steps{
+                sh('make check')
+            }
+        }
+        // avoid bundling devDependencies
+        stage('re-clean') {
+            steps {
+                sh('git clean -fdx')
+            }
+        }
+        stage('build image and upload') {
+            steps {
+                joyBuildImageAndUpload()
+            }
+        }
```

Comment:
This one should get the agentsshar launch, because of the marlin agent.

Reply:
Good point, I've added that. This is still just building the freestyle agentsshar job, rather than the fancy paramterized build of the new-Jenkins agents-installer job, but I thought it better to be consistent with other jobs for now so that we can fix them all together.